### PR TITLE
Add specs for data entry with Record Selector

### PR DESCRIPTION
**Pull request by seancolsen**
**Note:** This is stacked on top of #49 because I needed to make a small change to the cross-reference in the Navigation specs. I can unstack it if needed.

## What this PR does

- It adds more detail to the Record Selector design specs, describing the behavior of FK cells and inputs.

## Motivation

- @ghislaineguerin and @kgodey were both [wondering](https://github.com/centerofci/mathesar-wiki/pull/48#discussion_r930760727) about these details and I gave a bit of a hand-wavy response.
- For the Record Selector implementation, I need to figure this UX out soon -- and I've realized it's more challenging than I initially thought. There are lots of very subtle UX details and compromises to consider.

## Tips for review

- See the [rendered page](https://github.com/centerofci/mathesar-wiki/blob/linked_record_data_entry/design/specs/record-selector.md).
- The new content begins at _"Launch contexts"_ and ends immediately before _"Behavior after launch"_.

## Alternate designs considered

- Of the eight states presented in the mockup grid, I focused most on the design for the "Full" "Focused" "Cell" design because I felt that the others fell into place to some extent based on those decisions. To aid discussion within this PR, here are some alternates designs I considered for that state, with "A" being the winning design in this PR.

    image

    I welcome suggestions for alternate designs not represented above. Some of my rationale for choosing "A" is captured in the _"Design considerations and rationale"_ section of the page content. I have lots of other smaller thoughts along these lines that I can relay if needed.

### CODE REVIEW

**Comment by seancolsen** [comment]
@kgodey and @ghislaineguerin I'm requesting your reviews because this additional spec detail answers [questions](https://github.com/centerofci/mathesar-wiki/pull/48#discussion_r930760727) you both had.

**Comment by kgodey** [comment]
@seancolsen This looks good to me but I think I prefer B over A in your alternate designs, there's more contrast and it doesn't overload the summary pill. Why did you pick A over B?

**Comment by seancolsen** [comment]
Good question, @kgodey. I agree that B wins over A on aesthetics alone. And I'm certainly open to going with B, especially if Ghislaine prefers it too. I chose A because I felt that it won on usability for the following two reasons:

- For the input component, we need to display a "clear" icon (or figure out another way to clear the value).

    - If we put the "navigate" icon outside the pill (while maintaining consistency with its placement across the cell and input), then we have:

        image

    - Compare with "navigate" inside the pill (as I'm recommending)...

        image

    "Navigation inside the pill" seems slightly less confusing here because the styling helps to communicate that the "navigation" link is associated with the record whereas the "clear" button is associated with the input. I'll admit it's only a marginal improvement though, and aesthetics are compelling too!

- Also, (and perhaps more importantly) I really want to avoid the situation where the user sees the "navigate" icon and thinks its a "launch" icon. 

They might intend to open the record selector but instead navigate to the record page. This would be particularly bad if they're on a page with some ephemeral state because they could lose work by mistakenly navigating. I thought that putting the "navigate" icon within the pill would help to distinguish it from the "launch" icon.

Additional thoughts:

- Here's another idea for the input component...

    image

    I don't know, I don't love this aesthetically. It's too busy. But it does help to separate the function of the icons a bit more. Just brainstorming.

- If we do keep the "navigate" icon inside the pill, we could also play with the color...

    image

**Comment by kgodey** [comment]
I think in the interest of making a decision quickly, A is fine. I like the black color for the icon, but I assume @ghislaineguerin may make changes when she does styling anyway.

I don't love the clear icon either, but I think we should move forward and make changes later if needed.

**Review by kgodey — APPROVED** [review]

**Comment by seancolsen** [comment]
Thanks @kgodey. I'll wait to merge this until @ghislaineguerin has weighed in and also until #49 is merged.

File: design/specs.md [hunk 1]
```diff
@@ -32,6 +32,7 @@ These are specs for new features, corresponding to our [development milestones](
 - [Viewing Views](/design/specs/viewing-views-specs)
 
 ### Links between Tables (Foreign Keys, Table Relationships)
+- [Record Selector](/design/specs/record-selector.md)
```
[the remark this answers is not in the document]
Comment (kgodey) [reply design/specs.md:35]: I could have sworn I tried using `.md` links in the past and it didn't actually load the page. It looks like it is working, so nevermind.

[the remark this answers is not in the document]
Comment (seancolsen) [reply design/specs.md:35]: Cool

**Comment by kgodey** [comment]
I'm going to go ahead and merge this since @ghislaineguerin has had a week to review it.

**Pull request merged**